File: week 6/pset6/cash/cash.py
from cs50 import get_float

# declare values
q = 25
d = 10
n = 5
p = 1

# Ask the user for input
while True:
    money = get_float("Change owed: ")
    # validate the input
    if money > 0:
        break

# number of cents
cents = float(money * 100.0)

# Calculate the number of coins
coins = 0
while True:
    if q <= cents:
        cents = cents - q
        coins += 1
    elif d <= cents:
        cents = cents - d
        coins += 1
    elif n <= cents:
        cents = cents - n
        coins += 1
    elif p <= cents:
        cents = cents - p
        coins += 1

    if cents <= 0:
        break

# Display the number of coins
print(coins)

File: week 1/pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    //declare values
    int q = 25;
    int d = 10;
    int n = 5;
    int p = 1;

    //Ask the user for input
    float money;
    do
    {
        money = get_float("Change owed: ");
    }
    while (money < 0); //validate the input

    //convert the input from float to int
    int cents = round(money * 100);

    //Calculate the number of coins
    int coins = 0;
    do
    {
        if (q <= cents)
        {
            cents = cents - q;
            coins++;
        }
        else if (d <= cents)
        {
            cents = cents - d;
            coins++;
        }
        else if (n <= cents)
        {
            cents = cents - n;
            coins++;
        }
        else if (p <= cents)
        {
            cents = cents - p;
            coins++;
        }
    }
    while (cents > 0);

    //Display the number of coins
    printf("%i\n", coins);
}